### PR TITLE
feature(requirements.in): start using pip-tools to compile the requirements

### PR DESCRIPTION
**Pull request by fruch**
Since the introduction of newer pip versions `backtracing` feature is slowing
down the build of the docker image considerably

Using `pip-compile` we can have a specific list of the version we need, and
it would act as a lock file for the versions
```
pip-compile requirements.in > requirements.txt
```

* automaticlly do that on the `build_n_push.sh` docker/env/build_n_push.sh
* remove `requirements-python.txt` file
* `requirements.txt` is now activly the lock file

Ref: https://pip.pypa.io/en/stable/user_guide/#what-is-backtracking

## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I didn't leave commented-out/debugging code
- [x] I added the relevant `backport` labels
- ~~[ ] New configuration option are added and documented (in `sdcm/sct_config.py`)~~
- ~~[ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)~~
- [x] All new and existing unit tests passed (CI)
- [x] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

**Comment by fruch** [comment]
precommit is failing, should be fixed by #3075

File: README.rst [hunk 1]
```diff
@@ -115,7 +115,8 @@ To run SCT tests locally run following::
     # create a virtualenv for SCT
     pyenv virtualenv 3.8.3 sct38
     pyenv activate sct38
-    pip install -r requirements-python.txt
+    pip install -r requirements.txt
```
Comment (bentsi) [inline README.rst:118]: as you mentioned `requirements.txt` is now a lock file, does pip "understands" lockfile format?

**Review by bentsi — APPROVED** [review]

File: README.rst (hunk 1 above)
Comment (fruch) [reply README.rst:118]: yes, it plain `requirement.txt` format (that includes hashes), pip support it for quite some time now (few years)

File: requirements.in [hunk 2]
```diff
@@ -26,7 +26,7 @@ https://github.com/fruch/repodataParser/archive/py3.zip ; python_version > '3'
 pytest==4.6.4
 tcconfig==0.26.0
 selenium==3.141.0
-mysql-connector-python==8.0.19
+mysql-connector-python==8.0.20
```
Comment (fruch) [inline requirements.in:29]: this was change cause of a conflict in packages.
FYI our code that uses it doesn't really work anyhow, see #3069

**Pull request merged**